libcloud/common/azure.py

```diff
@@ -99,6 +105,96 @@ class AzureRawResponse(RawResponse):
     pass
 
 
+class AzureActiveDirectoryConnection(ConnectionUserAndKey):
+    """
+    Represents a single connection to Azure using Azure AD for Blob
+    """
+    conn_class = LibcloudConnection
+    driver = AzureBaseDriver
+    name = "Azure AD Auth"
+    responseCls = AzureResponse
+    rawResponseCls = AzureRawResponse
+
+    API_VERSION = "2017-11-09"
+
+    def __init__(
+        self,
+        key,
+        secret,
+        secure=True,
+        host=None,
+        port=None,
+        tenant_id=None,
+        identity=None,
+        cloud_environment=None,
+        **kwargs
+    ):
+        super(AzureActiveDirectoryConnection, self).__init__(identity, secret, **kwargs)
+        if not cloud_environment:
+            cloud_environment = "default"
```

Comment:
Would be better to just default ``cloud_environment`` argument in the method signature.